astroid/helpers.py

```diff
@@ -219,7 +219,7 @@ def is_supertype(type1, type2) -> bool:
     return _type_check(type1, type2)
 
 
-def class_instance_as_index(node):
+def class_instance_as_index(node) -> nodes.Const | None:
```

Comment:
```suggestion
def class_instance_as_index(node: bases.Instance) -> nodes.Const | None:
```

I did a quick check and this should work 😄